server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -119,6 +121,7 @@ public ReadOnlyEngine(EngineConfig config, SeqNoStats seqNoStats, TranslogStats
                 reader = wrapReader(reader, readerWrapperFunction);
                 searcherManager = new SearcherManager(reader, searcherFactory);
                 this.docsStats = docsStats(lastCommittedSegmentInfos);
+                this.translogStats = translogStats != null ? translogStats : translogStats(config, lastCommittedSegmentInfos);
```

Comment:
Can we assert that if `translogStats` is null then `obtainLock` is true? This ensures that we don't open two translog instances at the same time.

Reply:
++ especially as opening the translog here writes a new checkpoint file.

Reply:
This is a very good suggestion, thanks